ocos/kernels/vector_to_string.hpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+
+#include "kernels.h"
+#include "utils/string_utils.h"
+
+class VectorToStringImplBase
```

Comment:
You are already using the template Kernel, I wasn't seeing this base classes is needed, especially the virtual compute method

Reply:
Yep, the virtual base classes is unnecessary, I used them because let Op class(non-template) could have a template class, do you have any ideas on this? Sorry for my poor C++ knowledge.

Reply:
I didn't get your idea, why does the template has to be having a base class.
Just remove all staff of VectorToStringImplBase in your case.

Reply:
I'm sorry I didn't explain myself clearly. 
I want VectorToStringImpl to support int64/float/int32 input. VectorToString needs a map<key_type,value_type> to store the mapping relationship, so in this situation, the key_type should be a template type instead of a certain type. But in C++, a non-template class could not own a template class as its member, the following code should be wrong:
```c++
class A {
   std::hash_map<T, std::string> map_;
}
```
So I created a base class and put the template class members into the template class which is inherited from the base class.
This's the way I could only think of, I don't know whether you have a better idea.


However, it doesn't matter, ort-customop can't support multi-type input now. I decided to only support int64 input now, then the base class could be removed.